worker/provisioner/provisioner_task.go

```diff
@@ -1004,6 +1025,64 @@ func (task *provisionerTask) populateAvailabilityZoneMachines(ctx context.Provid
 	return nil
 }
 
+// checkProviderAvailabilityZones queries the known AZs.
+// If any are missing from the AZ-machines slice, add them.
+// If we have entries that are not known by the provider to be available zones,
+// check whether we have machines there.
+// If so, log a warning, otherwise we can delete them safely.
+func (task *provisionerTask) checkProviderAvailabilityZones(
+	ctx context.ProviderCallContext, zonedEnv providercommon.ZonedEnviron,
+) error {
+	azs, err := zonedEnv.AvailabilityZones(ctx)
```

Comment:
do we need a lock on this? there is no guarentee that the previous block of machines has completed provisioning before a new block is started. if provisioning retries within the loop, it goes back (?), to the distribution groups to find a new az?